core/node/eth_watch/src/event_processors/upgrades.rs

```diff
@@ -35,14 +34,13 @@ impl EventProcessor for UpgradesEventProcessor {
         storage: &mut Connection<'_, Core>,
         client: &dyn EthClient,
         events: Vec<Log>,
-    ) -> Result<(), Error> {
+    ) -> Result<(), EventProcessorError> {
         let mut upgrades = Vec::new();
-        for event in events
-            .into_iter()
-            .filter(|event| event.topics[0] == UPGRADE_PROPOSAL_SIGNATURE)
-        {
+        for event in events {
+            assert_eq!(event.topics[0], UPGRADE_PROPOSAL_SIGNATURE); // guaranteed by the watcher
```

Comment:
let's remove `UpgradesEventProcessor`, it was needed prior to some protocol version, now it's fully replaced by `GovernanceUpgradesEventProcessor`

Reply:
Just to confirm: We'll never need to process historic events for the old protocol versions, right? This functionality is not used by ENs, and I think we'll never need to re-process protocol upgrades on the main node.

Reply:
Yes